plugins/utils/which-key.nix

```
@@ -55,257 +265,308 @@ with lib;
+        z = defaultNullOpts.mkBool true "Bindings for folds, spelling and others prefixed with z";
+        g = defaultNullOpts.mkBool true "Bindings for prefixed with g";
       };
     };
 
-    operators = helpers.defaultNullOpts.mkAttrsOf types.str { gc = "Comments"; } ''
-      add operators that will trigger motion and text object completion
-      to enable all native operators, set the preset / operators plugin above
-    '';
+    win = {
+      no_overlap = defaultNullOpts.mkBool true "Don't allow the popup to overlap with the cursor.";
 
-    keyLabels = helpers.defaultNullOpts.mkAttrsOf types.str { } ''
-      override the label used to display some keys. It doesn't effect WK in any other way.
-    '';
+      border = defaultNullOpts.mkBorder "none" "which-key" "";
 
-    motions = {
-      count = helpers.defaultNullOpts.mkBool true "";
-    };
+      padding = defaultNullOpts.mkNullable (with types; listOf int) [
+        1
+        2
+      ] "Extra window padding [top/bottom, right/left]";
 
-    icons = {
-      breadcrumb = helpers.defaultNullOpts.mkStr "»" "symbol used in the command line area that shows your active key combo";
-      separator = helpers.defaultNullOpts.mkStr "➜" "symbol used between a key and it's label";
-      group = helpers.defaultNullOpts.mkStr "+" "symbol prepended to a group";
-    };
+      title = defaultNullOpts.mkBool true "Whether to show the title.";
 
-    popupMappings = {
-      scrollDown = helpers.defaultNullOpts.mkStr "<c-d>" "binding to scroll down inside the popup";
-      scrollUp = helpers.defaultNullOpts.mkStr "<c-u>" "binding to scroll up inside the popup";
-    };
+      title_pos = defaultNullOpts.mkStr "center" "Position of the title";
 
-    window =
-      let
-        spacingOptions = types.submodule {
-          options =
-            genAttrs
-              [
-                "top"
-                "right"
-                "bottom"
-                "left"
-              ]
-              (
-                n:
-                mkOption {
-                  type = types.ints.unsigned;
-                  description = "Spacing at the ${n}.";
-                }
-              );
-        };
-      in
-      {
-        border = helpers.defaultNullOpts.mkBorder "none" "which-key" "";
-        position = helpers.defaultNullOpts.mkEnumFirstDefault [
-          "bottom"
-          "top"
-        ] "";
-        margin = helpers.defaultNullOpts.mkNullable spacingOptions {
-          top = 1;
-          right = 0;
-          bottom = 1;
-          left = 0;
-        } "extra window margin";
-        padding = helpers.defaultNullOpts.mkNullable spacingOptions {
-          top = 1;
-          right = 2;
-          bottom = 1;
-          left = 2;
-        } "extra window padding";
-        winblend = helpers.defaultNullOpts.mkNullable (types.ints.between 0
+      zindex = defaultNullOpts.mkInt 1000 "Layer depth on the popup window";
```

```suggestion
      zindex = defaultNullOpts.mkInt 1000 "Layer depth on the popup window.";
```

Are negative numbers allowed, or should this be `mkUnsignedInt`?

---

Will have to double check. Similar to padding, I was thinking negatives like in web development.

---

Which-key throws error on negative. Changing.